### DISCUSSION TITLE
Protecting API routes on a separate backend (currently using database sessions)

### DISCUSSION THREAD

**Discussion by daxidngyn** [nextauthjs/next-auth#4863] [Help]
### Question 💬

I currently have the following setup in a monorepo:
- Next.js frontend with next-auth. I am storing sessions in my MongoDB database using a Prisma adapter.
- Express.js backend.

My authentication functionality is currently working, but I wish to secure the API routes on my express server and authorize users. At first, I tried sending the `sessionId` to the server to validate with my database, but upon further research found that this wasn't a secure method.

From my limited understanding, I have to instead generate a JWT token on the server and pass this back to the client, storing it in the session. I can then use the NextAuth.js session to retrieve this token and make following requests. Is this correct? If so, how do I go about generating this token?

In addition, the majority of articles I've stumbled across while researching this topic were using JWT instead of database sessions, should I be doing so instead?

### How to reproduce ☕️

`[...nextauth].ts:`
```
import NextAuth, { NextAuthOptions } from "next-auth";
import GoogleProvider from "next-auth/providers/google";
import DiscordProvider from "next-auth/providers/discord";
import { PrismaAdapter } from "@next-auth/prisma-adapter";

import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

export const authOptions: NextAuthOptions = {
  adapter: PrismaAdapter(prisma),
  providers: [
    GoogleProvider({
      //@ts-ignore
      clientId: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID,
      //@ts-ignore
      clientSecret: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_SECRET,
      profile(profile) {
        return {
          id: profile.sub,
          email: profile.email,
          profileImg: profile.picture,
        };
      },
    }),
    DiscordProvider({
      //@ts-ignore
      clientId: process.env.NEXT_PUBLIC_DISCORD_CLIENT_ID,
      //@ts-ignore
      clientSecret: process.env.NEXT_PUBLIC_DISCORD_CLIENT_SECRET,
      profile(profile) {
        return {
          id: profile.id,
          username: profile.username,
          email: profile.email,
          profileImg: `https://cdn.discordapp.com/avatars/${profile.id}/${profile.avatar}`,
          bannerImg: profile.banner
            ? `https://cdn.discordapp.com/banners/${profile.id}/${profile.banner}`
            : null,
        };
      },
    }),
  ],
  callbacks: {
    async signIn({ user, account, profile }) {
      return true;
    },
    // @ts-ignore
    async session({ session, user }) {
      return Promise.resolve({
            expires: session.expires,
            id: user.id,
            email: user.email,
            walletAddress: user.walletAddress,
            username: user.username,
            verified: user.emailVerified,
            fanPoints: user.totalFanPoints,
      });
    },
  },
};

export default NextAuth(authOptions);
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by zulfio** [comment]
Hi @daxidngyn, did you find the solution to protecting your separate backend with next.js? I'm also facing the same issue
